osu.Game.Modes.Mania/UI/ManiaHitRenderer.cs

```diff
@@ -12,7 +13,8 @@ public class ManiaHitRenderer : HitRenderer<ManiaBaseHit>
     {
         private readonly int columns;
 
-        public ManiaHitRenderer(int columns = 5)
+        public ManiaHitRenderer(Beatmap beatmap, int columns = 5)

```

Comment:
Definitely re: when mania is implemented.

Actually very interesting point re: workingbeatmap, because then the hitrenderer could also do Apply/Remove for each mod on the hitobjects internally, along with perhaps moving the current auto mod application inside too.